### PR TITLE
Fix for STORE-1286: Fix store asset api type validation issue

### PR DESCRIPTION
**Pull request by tmkasun**
In this PR:

* pass `tenantId` with `validateType` method to check the validity of the API endpoint name
* update getTenantId method to return tenantId based on user domain if domain name available with the request
* Change the 404 error message return type from HTML to JSON in API calls 

Addresses the following issue: [STORE-1286](https://wso2.org/jira/browse/STORE-1286)

### CODE REVIEW

File: apps/store/controllers/assets-router.jag [hunk 1]
```diff
@@ -40,8 +41,10 @@ var mapper = function (path) {
 
 var handleApiEndpoint = function (request, session, uriParams) {
     var type = uriParams.type;
-    if(!validateType(type)){
-        response.sendError(404, 'Unabled to locate a active endpoint for the type: ' + type);
+    var tenantId = getTenantId(request, session);
+    if(!validateType(type, tenantId)){
```
Comment (daneshk) [inline apps/store/controllers/assets-router.jag]: formatting issue. please check

File: apps/store/controllers/assets-router.jag [hunk 2]
```diff
@@ -139,9 +141,20 @@ var validateType = function (type, tenantId){
  * @return {[type]}         Tenant id for the request
  */
 var getTenantId = function(request, session) {
+    var server = require('carbon').server;
     var tenantAPI = require('/modules/tenant-api.js').api;
-    var tenantDetails = tenantAPI.createTenantDetails(request, session);
-    var tenantId = tenantDetails.tenantId;
+    var tenantId;
+    var domain = request.getHeader("domain");
+    if(!domain){
```
Comment (daneshk) [inline apps/store/controllers/assets-router.jag]: formatting issue. please check

File: apps/store/controllers/assets-router.jag (hunk 1 above)
Comment (tmkasun) [inline apps/store/controllers/assets-router.jag]: Thanks @daneshk , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/842/commits/d825d8c011cf7d57ddf865fadb74847d5f3fa9fd)

File: apps/store/controllers/assets-router.jag (hunk 2 above)
Comment (tmkasun) [inline apps/store/controllers/assets-router.jag]: Thanks @daneshk , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/842/commits/d825d8c011cf7d57ddf865fadb74847d5f3fa9fd)

**Pull request merged**